tournament/src/api/views.py

```diff
@@ -0,0 +1,130 @@
+        if max_players is not None:
+            tournament.max_players = max_players
+        registration_deadline = json_request.get('registration-deadline')
+        if json_request.get('registration-deadline') is not None:
+            registration_deadline = parser.isoparse(registration_deadline)
+            tournament.registration_deadline = TournamentView.convert_to_utc_datetime(registration_deadline)
+        tournament.save()
+        return JsonResponse(model_to_dict(tournament), status=201)
+
+    @staticmethod
+    def is_valid_tournament(json_request: Dict[str, Any]) -> Tuple[bool, list[str]] | Tuple[bool, None]:
+        errors = []
+        name = json_request.get('name')
+        max_players = json_request.get('max-players')
+        registration_deadline = json_request.get('registration-deadline')
+        is_private = json_request.get('is-private')
+
+        valid_name, name_errors = TournamentView.is_valid_name(name)
+        valid_max_players, max_players_error = TournamentView.is_valid_max_players(max_players)
+        valid_deadline, deadline_error = TournamentView.is_valid_deadline(registration_deadline)
+        valid_private, is_private_error = TournamentView.is_valid_private(is_private)
+
+        if not valid_name:
+            errors.append(name_errors)
+        if not valid_max_players:
+            errors.append(max_players_error)
+        if not valid_deadline:
+            errors.append(deadline_error)
+        if not valid_private:
+            errors.append(is_private_error)
+
+        if errors:
+            return False, errors
+        return True, None
+
+    @staticmethod
+    def is_valid_name(name: Any) -> tuple[bool, list[str]] | tuple[bool, None]:
+        errors = []
+
+        if name is None:
+            errors.append('Missing name field')
+        if len(name) < settings.MIN_TOURNAMENT_NAME_LENGTH:
+            errors.append(f'Tournament name must contain at least {settings.MIN_TOURNAMENT_NAME_LENGTH} characters')
+        elif len(name) > settings.MAX_TOURNAMENT_NAME_LENGTH:
+            errors.append(f'Tournament name must contain less than {settings.MAX_TOURNAMENT_NAME_LENGTH} characters')
+        if not name.replace(' ', '').isalnum():
+            errors.append('Tournament name may only contain letters, numbers and spaces')
+
+        if errors:
+            return False, errors
+        return True, None
+
+    @staticmethod
+    def is_valid_max_players(max_players: Any) -> tuple[bool, str | None]:
+        if max_players is None:
+            return True, None
+        if not isinstance(max_players, int):
+            return False, 'Max players must be an integer'
+        if max_players > settings.MAX_PLAYERS:
+            return False, f'Max players must contain less than {settings.MAX_PLAYERS} slots'
+        if max_players < settings.MIN_PLAYERS:
+            return False, f'Max players must contain at least {settings.MIN_PLAYERS} slots'
```

Comment:
Error message is incorrect you should return either:
`Tournament must contain less than {settings.MAX_PLAYERS} slots`
or:
`Max players must be less than {settings.MAX_PLAYERS}`